java/src/jmri/jmrix/mqtt/MqttConsistManager.java

```diff
@@ -0,0 +1,122 @@
+     * consist reader thread, which retrieves consist information from the
+     * command station.
+     *
+     * @param memo the associated connection memo
+     */
+    public MqttConsistManager(MqttSystemConnectionMemo memo) {
+        super();
+        adapterMemo = memo;
+    }
+
+    public void setSendTopic(@Nonnull String sendTopicPrefix) {
+        this.sendTopicPrefix = sendTopicPrefix;
+    }
+
+    @Nonnull
+    public String sendTopicPrefix = "cab/$address/consist";
+
+
+    /**
+     * This implementation does not support advanced consists, so return true.
+     *
+     */
+    @Override
+    public boolean isCommandStationConsistPossible() {
+        return true;
+    }
+
+    /**
+     * Does a CS consist require a separate consist address? CS consist
+     * addresses are assigned by the user, so return true.
+     *
+     */
+    @Override
+    public boolean csConsistNeedsSeperateAddress() {
+        return true;
+    }
+
+    /**
+     * Add a new MQTT Consist with the given address to
+     * consistTable/consistList.
+     */
+    @Override
+    public Consist addConsist(LocoAddress address) {
+        if (! (address instanceof DccLocoAddress)) {
+            throw new IllegalArgumentException("address is not a DccLocoAddress object");
+        }
+        if (consistTable.containsKey(address)) { // no duplicates allowed across all connections
+            return consistTable.get(address);
+        }
+        MqttConsist consist;
+        consist = new MqttConsist((DccLocoAddress) address, (MqttSystemConnectionMemo) adapterMemo,
+                sendTopicPrefix);
+        consistTable.put(address, consist);
+        notifyConsistListChanged();
+        return consist;
+    }
+
+    /* Request an update from the layout, loading
+     * Consists from the command station.
+     */
+    @Override
+    public void requestUpdateFromLayout() {
+    }
+
+    @Override
+    protected boolean shouldRequestUpdateFromLayout() {
+        return false;
+    }
+
+    /**
+     * If a consist exists with the given address, the consist is activated on the controller,
+     * otherwise it does nothing.
+     * This is used by a throttle in case it is controlling a consist.
+     */
+    public void activateConsist(LocoAddress address) {
+
+        if (!consistTable.containsKey(address)) return;
+
+        ((MqttConsist)consistTable.get(address)).activate();
+
+    }
+
+    /**
+     * If a consist exists with the given address, the consist is deactivated on the controller,
+     * otherwise it does nothing.
+     * This is used by a throttle in case it is controlling a consist.
+     */
+    public void deactivateConsist(LocoAddress address) {
+
+        if (!consistTable.containsKey(address)) return;
+
+        ((MqttConsist)consistTable.get(address)).deactivate();
+
+    }
+
+
+
+    private final static Logger log = LoggerFactory.getLogger(MqttConsistManager.class);
```

Comment:
Static analysis failes on this line due to that the variable `log` is not used. I suggest you comment this line since it's useful to have if it's needed later.

Reply:
Fixed